src/main/java/cloud/eppo/Configuration.java

```diff
@@ -0,0 +1,203 @@
+  }
+
+  /**
+   * Builder to create the immutable config object.
+   *
+   * @see cloud.eppo.Configuration for usage.
+   */
+  public static class Builder {
+    private static final ObjectMapper mapper =
+        new ObjectMapper().registerModule(EppoModule.eppoModule());
+
+    private final boolean isConfigObfuscated;
+    private final Map<String, FlagConfig> flags;
+    private Map<String, BanditReference> banditReferences;
+    private Map<String, BanditParameters> bandits = Collections.emptyMap();
+
+    public static Configuration emptyConfig() {
+      return new Configuration(
+          Collections.emptyMap(), Collections.emptyMap(), Collections.emptyMap(), false);
+    }
+
+    public Builder(String flagJson) {
+      this(flagJson, false);
+    }
+
+    public Builder(String flagJson, boolean isConfigObfuscated) {
+      this.isConfigObfuscated = isConfigObfuscated;
+
+      if (flagJson == null || flagJson.isEmpty()) {
+        throw new RuntimeException(
+            "Null or empty configuration string. Call `Configuration.Builder.Empty()` instead");
+      }
+
+      // Build the flags config from the json string.
+      FlagConfigResponse config;
+      try {
+        config = mapper.readValue(flagJson, FlagConfigResponse.class);
+      } catch (JsonProcessingException e) {
+        throw new RuntimeException(e);
+      }
+
+      if (config == null || config.getFlags() == null) {
+        log.warn("Flags missing in ufc response");
+        flags = Collections.emptyMap();
+      } else {
+        flags = Collections.unmodifiableMap(config.getFlags());
+        banditReferences = Collections.unmodifiableMap(config.getBanditReferences());
+        log.debug("Loaded {} flag configs from configuration response", flags.size());
+      }
+    }
+
+    public boolean requiresBanditModels() {
+      Set<String> neededModelVersions = referencedBanditModelVersion();
+      return !loadedBanditModelVersions().containsAll(neededModelVersions);
+    }
+
+    public Set<String> loadedBanditModelVersions() {
+      return bandits.values().stream()
+          .map(BanditParameters::getModelVersion)
+          .collect(Collectors.toSet());
+    }
+
+    public Set<String> referencedBanditModelVersion() {
+      return banditReferences.values().stream()
+          .map(BanditReference::getModelVersion)
+          .collect(Collectors.toSet());
+    }
+
+    public Builder banditParametersFrom(Configuration currentConfig) {
+      if (currentConfig == null || currentConfig.bandits == null) {
+        bandits = Collections.emptyMap();
+      } else {
+        bandits = currentConfig.bandits;
+      }
+      return this;
+    }
+
+    public Builder banditParameters(String banditParameterJson) {
+      BanditParametersResponse config;
+      try {
+        config = mapper.readValue(banditParameterJson, BanditParametersResponse.class);
+      } catch (JsonProcessingException e) {
+        log.error("Unable to parse bandit parameters response");
```

Comment:
here too we may want to weed out the word "response"

Reply:
This is a really good nit :). "response" is like 98% correct here because we're parsing a string into a `BanditParametersResponse` object, but the string may not be a "response" per se and we want to convey a local context to the dev/logs and in that local context it's a JSON string. 🧑‍🍳